### ISSUE TITLE
--force-sourcecode option doesn't work with remote runs (using either restart or proto)

### ISSUE THREAD

**Issue by ghwatson** [guildai/guildai#260]
I edited a file and tried to restart a prior run on a remote (I also tried using proto). Wanting to use the current sourcecode, I used the flag for that. Whether I sent the command to a remote, ssh'd in and tried to restart there, or tried to restart locally I get the error:

```
guild: cannot find definition for operation 'gpkg.anonymous-c01716eb/eval_1' in run e78ae1237b8d4b19891f487c7004405f
The definition is required when setting flags for start or restart.
```

As a sanity check, I tried using the --force-sourcecode with a run that was run locally and I didn't get this error.

As a workaround in the meantime I'm just doing the run again from scratch. Let me know if I can provide any more information!

**Comment by gar1t** [comment]
Thanks for the report! This is a design oversight - we never properly implemented remote support for this scenario. The correct behavior short term should be to fail with a message stating that `--force-sourcecode` is not supported for remote operations - punting the problem but at least being clear about it. Long term this needs to be supported.

For the record here, with `--restart`, `--remote`, and `--force-sourcecode`, I believe the correct behavior is that Guild repackages the remote operation and deploys to the restart run location.

The options `--proto`, `--remote`, and `--force-sourcecode` should be a standard operation that uses the proto run flag values as default.

Work around for proto should be straight forward - just manually look up the flags you want to use and specify what you want directly.

To work around restart, that's much harder. Everything I can think of is unacceptably complicated to the point I don't even want to suggest them :)

I'll see if I can get this fixed sooner than later as it really ought to be working in the first place. Sorry about that!

**Comment by ghwatson** [comment]
Yep all good regarding the workaround, thanks! Fortunately I'm not relying on warm starts this so no serious need - manual workaround good enough.
